ibis/expr/operations.py

```diff
@@ -1086,7 +1086,7 @@ def count(self):
 
 class CountDistinct(Reduction):
 
-    input_type = [rules.column]
+    input_type = [rules.column, boolean(name='where', optional=True)]
```

Comment:
I'm going to fix this in a PR soon. It requires more than just adding this argument to the operation. Please revert this change.